autoload/compe/confirmation.vim

```diff
@@ -0,0 +1,49 @@
+let s:Position = vital#compe#import('VS.LSP.Position')
+let s:TextEdit = vital#compe#import('VS.LSP.TextEdit')
+let s:CompletionItem = vital#compe#import('VS.LSP.CompletionItem')
+
+"
+" compe#confirmation#lsp
+"
+function! compe#confirmation#lsp(args) abort
+  let l:completed_item = a:args.completed_item
+  let l:completion_item = a:args.completion_item
+  let l:current_position = s:Position.cursor()
+  let l:suggest_position = { 'line': line('.') - 1, 'character': l:current_position.character - strchars(l:completed_item.word) }
+  let l:request_position = a:args.request_position
+  call s:CompletionItem.confirm({
+  \   'suggest_position': l:suggest_position,
+  \   'request_position': l:request_position,
+  \   'current_position': l:current_position,
+  \   'current_line': getline('.'),
+  \   'completion_item': l:completion_item,
+  \   'expand_snippet': function('s:expand_snippet'),
+  \ })
+endfunction
+
+"
+" get_snippet_expanding_function
+"
+function! s:expand_snippet(args) abort
+  if exists('g:loaded_vsnip')
+    call vsnip#anonymous(a:args.body)
+  else
```

Comment:
If someone wants to support ultisnips/snippets.nvim, feel free to create PR.

Reply:
May I ask for a help? I'm trying to implement support for snippets.nvim but I got stuck :laughing:
I managed to setup the source and one of the item looks like this.

```lua
{
  word = "todo",
  userdata = {
    snippets = {
      snippet = {"TODO: $0"},
    }
  },
  kind = "Snippet",
  abbr = "todo",
  dup = 1
}
```

Snippets.nvim has this function called `expand_at_cursor` which will expand the snippet. How do I tell compe to do something like `expand_at_cursor("TODO: $0")`?

I tried adding this but I'm not sure what `a:args.body` is.

```viml
function! s:expand_snippet(args) abort
  if exists('g:loaded_vsnip')
    call vsnip#anonymous(a:args.body)
  else
    " TODO: find a way to check if snippets.nvim is available
    " call s:simple_expand_snippet(a:args.body)
    call luaeval('require"snippets".expand_at_cursor("'. a:args.body .'")')
  endif
endfunction
```

Thanks in advance!

Reply:
The `a:args.body` is string.

If `require"snippets".expand_at_cursor()` accepts string, You should use `call luaeval('require"snippets".expand_at_cursor(_A[1])', [a:args.body])` intead.

Reply:
Hmm, I tried that but it didn't work. Also, vsnip stopped working apparently, it no longer expands the snippet. I've tried to reproduce this using [this init.vim](https://paste.sh/iXzU7U24#BbpwaK6IivtKT5EIFL-FPHEp) then add a snippet with this content. 

```json
{
  "todo": {
    "prefix": ["todo"],
    "body": [
      "TODO: $0"
    ],
    "description": "Todo reminder"
  }
}
```

The snippet doesn't get expanded when I press `<CR>`

Reply:
Thank you for your report.
I've fixed the vsnip source expansion.